.github/workflows/pytest.yml

```diff
@@ -0,0 +1,45 @@
+name: pytest (RHEL 8, RHEL 9)
+
+on:
+  pull_request:
+    paths:
+      - "src/*"
+
+jobs:
+  pytest:
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - name: "Fedora Rawhide"
+            image: "registry.fedoraproject.org/fedora:rawhide"
+          - name: "Fedora Latest"
+            image: "registry.fedoraproject.org/fedora:latest"
+          - name: "CentOS Stream 9"
+            image: "quay.io/centos/centos:stream9"
+          - name: "CentOS Stream 8"
+            image: "quay.io/centos/centos:stream8"
+
+    runs-on: "ubuntu-latest"
+    container:
+      image: ${{ matrix.image }}
+
+    steps:
+      - name: "Checkout the repository"
+        uses: actions/checkout@v4
+
+      - name: "Install dependencies"
+        # Setuptools is required because we still rely on 'distutils' which have been dropped in Python 3.12
```

Comment:
hm considering the client still needs distutils, wouldn't it be better to keep it in `requirements.txt` for now? this avoids having the dependencies in two different places (`requirements.txt` and here in this workflow)

in the end, once the distutils usage is dropped, also `requirements.txt` can be cleaned